packages/nuxt/src/core/nitro.ts

```diff
@@ -232,6 +232,16 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
     handler: resolve(distDir, 'core/runtime/nitro/renderer')
   })
 
+  if (nuxt.options.experimental.noVueServer) {
+    nitro.hooks.hook('rollup:before', (nitro) => {
+      if (nitro.options.preset === 'nitro-prerender') { return }
+      nitro.options.handlers.shift()
```

Comment:
This seems little bit unsafe workaround.

I think we might support `experimental.noRenderer` directly to nitro handling this case. (and disable the flag with `prerender`

Reply:
As this is an experimental option, what do you think about merging this as is, and then switch over to `noRenderer` once nitro supports it?

Reply:
Looks good to me.